src/saturn_engine/worker/executors/queue.py

```diff
@@ -45,6 +55,44 @@ def start(self) -> None:
         self.submit_tasks.start()
         self.processing_tasks.start()
 
+    async def _process_error(
+        self, processable: ExecutableMessage, exception: Exception
+    ):
+        # Match the exception and consume it if found, else reraise the exception
+        for k, ts in processable.output.items():
+            # TODO: better way to identify error output, maybe seperate field instead
+            if is_output_error_handler(k):
+                exc_filter = parse_warning_filter(k, escape=False)
+                if exc_filter.action == "ignore":
+                    raise HandledException("Exception ignored")
+
+                exc_details = get_exception_details()
+                if does_output_match(exc_details, exc_filter):
+                    self.consuming_tasks.create_task(
+                        self.consume_output(
+                            processable=processable,
+                            output=[
+                                PipelineOutput(
+                                    channel=k,
+                                    message=TopicMessage(
+                                        args=asdict(
+                                            ErrorMessageArgs(
+                                                exception_type=exception.__class__.__name__,
+                                                exception_module=exception.__class__.__module__,
+                                                exception_message=str(exception),
+                                            )
+                                        )
+                                    ),
+                                )
+                                for t in ts
+                            ],
+                        ),
+                        name=f"consume-error({processable})",
+                    )
+                    # Do we want multiple rules to match in order to consume an exception more than once?
+                    raise HandledException("Exception has been consumed")
+        raise exception
```

Comment:
This reset the exception context (losing traceback and all).

I would suggest returning `None` and check from the caller

```python
outputs = self._process_error(...)
if output is None:
  raise
```

Reply:
Changed the logic a bit, see `bootstrap.py:run_pipeline`